src/lib/arch/riscv/Instruction_decode.cc

```diff
@@ -19,6 +19,24 @@ namespace riscv {
 Register csRegToRegister(unsigned int reg) {
   // Check from top of the range downwards
 
+  // Metadata could produce either 64-bit floating point register or 32-bit
+  // floating point register. Map both encodings to the same SimEng register.
+  // Only 64-bit registers are supported
+
+  // Modulus ensures only 64 bit registers are recognised
+  if (RISCV_REG_F31_64 >= reg && reg >= RISCV_REG_F0_64 && reg % 2 == 0) {
+    // Register ft0.64 has encoding 34 with subsequent encodings interleaved
+    // with 32 bit floating point registers. See riscv.h
+    return {RegisterType::FLOAT, static_cast<uint16_t>((reg - 34) / 2)};
+  }
+
+  // Modulus ensures only 32 bit registers are recognised
+  if (RISCV_REG_F31_32 >= reg && reg >= RISCV_REG_F0_32 && reg % 2 == 1) {
+    // Register ft0.32 has encoding 33 with subsequent encodings interleaved
+    // with 64 bit floating point registers. See riscv.h
+    return {RegisterType::FLOAT, static_cast<uint16_t>((reg - 33) / 2)};
+  }
```

Comment:
when doing `reg-34` etc, does the `34` represent the enum value of `ft0.64`? If so then it is best to use the actual enum name (`RISCV_REG_F31_64`). 

Also, how can you ensure that dividing by 2 will always be a whole number before you static cast it to a uint?

Reply:
Enum encoding used.

The last clause of each ```if``` requires ````reg``` to be either odd OR even. In the even case we subtract 34; even - even = even and so even / 2 is whole. In the odd case we subtract 33; odd - odd = even and so even / 2 is whole. So in both cases the output of the computation is whole before being cast to uint

Reply:
OK understood. Could you add a brief comment somewhere here with this same information to avoid any confusion or doubt in the future?